cardinal/world.go

```diff
@@ -207,6 +208,35 @@ func NewMockWorld(opts ...WorldOption) (*World, error) {
 	return world, nil
 }
 
+func RegisterMessage[In any, Out any](world *World, name string, opts ...message.MessageOption[In, Out]) error {
```

Comment:
This should be in `cardinal/cardinal.go` where we keep all of our other `Register<...>` functions at. The whole idea here is to have exactly 1 way to do something so the API surface area is as narrow as possible and you don't have random function lying around everywhere.

Reply:
Just note, right now there's multiple ways of registering messages. (The old way still works) I'm going to change this incrementally with multiple PRs to keep things readable and manageable. I will be slowly changing it across multiple PRs to the new way and once all dependencies are resolved I will remove everything associated with the "old way"

And yes I will move this to cardinal.go.